tracer/src/Datadog.Trace/Activity/Handlers/DefaultActivityHandler.cs

```diff
@@ -102,18 +108,26 @@ public void ActivityStarted<T>(string sourceName, T activity)
                     return;
                 }
 
-                ActivityMappingById.GetOrAdd(activity.Id, _ => new(activity.Instance, CreateScopeFromActivity(activity, parent, traceId, spanId, rawTraceId, rawSpanId)));
+                Log.Information("GetOrAdd for {Name} with ID: {ID}", activity.OperationName, activity.Id);
+
+                if (isRemoteContext)
+                {
+                    ActivityMappingById.GetOrAdd(activity.ParentId!, _ => new(activity.Instance, CreateScopeFromActivity(activity, parent, traceId, spanId, rawTraceId, rawSpanId)));
+                }
+                else
+                {
+                    ActivityMappingById.GetOrAdd(activity.Id, _ => new(activity.Instance, CreateScopeFromActivity(activity, parent, traceId, spanId, rawTraceId, rawSpanId)));
+                }
             }
             catch (Exception ex)
             {
                 Log.Error(ex, "Error processing the OnActivityStarted callback");
             }
 
-            static Scope CreateScopeFromActivity(T activity, SpanContext? parent, ulong? traceId, ulong? spanId, string? rawTraceId, string? rawSpanId)
+            static Scope CreateScopeFromActivity(T activity, ISpanContext? parent, ulong? traceId, ulong? spanId, string? rawTraceId, string? rawSpanId)
```

Comment:
I don't think this change is necessary?

Reply:
That's correct, I think I forgot to undo it